src/ert/_c_wrappers/enkf/ert_config.py

```diff
@@ -319,7 +338,11 @@ def read_forward_model(
             if args:
                 job.private_args = SubstitutionList()
                 try:
-                    job.private_args.add_from_string(args)
+                    if isinstance(args, str):
+                        job.private_args.add_from_string(args)
+                    else:
+                        for key, val in args:
+                            job.private_args.addItem(key, val)
```

Comment:
please add a comment for this special handling, which is due to old/new parser, so we can remove the special handling once we switch to new parsa